db/blip_handler.go

```diff
@@ -149,9 +149,12 @@ func collectionBlipHandler(next blipHandlerFunc) blipHandlerFunc {
 				DatabaseCollection: bh.db.singleCollection,
 				user:               bh.db.user,
 			}
+			if bh.nonCollectionAwareContext == nil {
+				bh.nonCollectionAwareContext = &BlipSyncCollectionContext{dbCollection: bh.collection.DatabaseCollection}
```

Comment:
This seems to be racing in Jenkins. I think because two BLIP messages mean this handler can be called simultaneously.

Not sure what the correct course of action is here without a deeper look.

Reply:
Wow, I didn't see that at all, and it didn't repro for me at all, thanks for pointing it out.